pkg/log/log.go

```diff
@@ -27,40 +27,66 @@ type Logger struct {
 
 // Event represents a log event, mimicking zerolog.Event.
 type Event struct {
-	logger  *Logger
-	level   slog.Level
-	attrs   []slog.Attr
-	isPanic bool
+	logger   *Logger
+	level    slog.Level
+	attrs    []slog.Attr
+	isPanic  bool
+	caller   string
+	function string
+}
+
+// newEvent creates a new log event with caller information captured at the point of creation.
+func (l Logger) newEvent(level slog.Level, isPanic bool) *Event {
+	var caller, function string
+	// Get the program counter at the caller's location
+	pc := make([]uintptr, 1)
+	// Skip: runtime.Callers(0), newEvent(1), Info/Error/etc(2), actual caller(3)
+	n := runtime.Callers(3, pc)
```

Comment:
Maybe
```suggestion
	n := runtime.Callers(callerSkipFrameCount, pc)
```
With callerSkipFrameCount being 3

Reply:
yes adjusted as such.